File: DP/2293_동전 1/2293_동전 1.py
import sys;
input = sys.stdin.readline
n, k = map(int, input().split())

dp = [0 for i in range(k+1)]
dp[0] = 1

for _ in range(n):
    coin = int(input())
    for j in range(coin, k+1):
        dp[j] = dp[j] + dp[j-coin]

print(dp[k])

File: DP/2293_동전 1/2293_동전 1.js
const [N, K, ...coins] = `3 10
1
2
5`
  .split(/\s+/)
  .map(Number);

const DP = Array(K + 1).fill(0);
DP[0] = 1;

for (const coin of coins) {
  for (let i = coin; i <= K; i++) {
    DP[i] = DP[i] + DP[i - coin];
  }
}

console.log(DP[K + 1]);
